multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/executor/CancelExpiredTasksThread.java

```diff
@@ -0,0 +1,65 @@
+package org.cloudfoundry.multiapps.controller.process.executor;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.util.ConcurrentReferenceHashMap;
+
+import java.util.Map;
+import java.util.concurrent.Future;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class CancelExpiredTasksThread extends Thread {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CancelExpiredTasksThread.class);
+    private Map<Future<?>, Long> tasksStartTime = new ConcurrentReferenceHashMap<>();
```

Comment:
Is it possible to keep the task reference by some property of the task, not the whole object?

Reply:
nothing usable is provided by the future api